test/tools/h3_request/h3_request.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import asyncio
+import sys
+from typing import AsyncIterator, cast
+from urllib.parse import urlparse
+
+import aioquic
+from aioquic.asyncio.protocol import QuicConnectionProtocol
+from aioquic.h3.connection import H3_ALPN, H3Connection
+from aioquic.h3.events import (
+    DataReceived,
+    H3Event,
+    HeadersReceived,
+)
+from aioquic.quic.events import QuicEvent
+
+CA_PATH = "test/config/integration/certs/cacert.pem"
+
+
+class Http3Client(QuicConnectionProtocol):
+    """Note, this class is extremely minimal.
+
+    It supports only GET, doesn't properly validate URLs, etc. Since this
+    is just for tests, that's all that's required right now.
+    It is based on https://github.com/aiortc/aioquic/blob/main/examples/http3_client.py
+    which is a far more complete implementation.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._http = H3Connection(self._quic)
+        self._stream_ids: Dict[int, asyncio.Future[bool]] = {}
+
+    def headers_received(self, event: H3Event) -> None:
+        if not self._include_headers:
+            return
+        for header, value in event.headers:
+            print(f"{header.decode('utf-8')}: {value.decode('utf-8')}\n", end="")
+        print("\n", end="")
+
+    def http_event_received(self, event: H3Event) -> None:
+        stream_id = event.stream_id
+        if stream_id not in self._stream_ids:
+            return
+        if isinstance(event, HeadersReceived):
+            self.headers_received(event)
+        elif isinstance(event, DataReceived):
+            print(event.data.decode("utf-8"), end="")
+        else:
+            raise Exception(f"unexpected quic event type {event}")
```

Comment:
```suggestion
            raise QuicClientException(f"unexpected quic event type {event}")
```

better not to raise `Exception` as you cant then catch it specifically

Reply:
(you will need to define the suggestion or use an existing one)